backend/backend/RoomManager.c

```diff
@@ -403,27 +488,334 @@ BOOL StartGame(_Inout_ PCONNECTION_INFO pConnInfo)
 
 BOOL PlayerSelectTeam(_Inout_ PCONNECTION_INFO pConnInfo, _In_ UINT TeamMemberCnt, _In_ UINT32 TeamMemberList[])
 {
-    return TRUE;
+    PGAME_ROOM pRoom = pConnInfo->pRoom;
+    BOOL bSuccess = FALSE;
+    if (!pRoom)
+        return ReplyPlayerSelectTeam(pConnInfo, FALSE, "You are not in a room.");
+    AcquireSRWLockExclusive(&pRoom->PlayerListLock);
+    __try {
+
+        if (!pRoom->bGaming)
+        {
+            bSuccess = ReplyPlayerSelectTeam(pConnInfo, FALSE, "Game hasn't started yet.");
+            __leave;
+        }
+        // TODO: ��鵱ǰ�Ƿ��Ƕӳ�ѡ���Ա����Ϸ�׶�
+
+        if (pRoom->Game_Win>=3)
+            return ReplyPlayerSelectTeam(pConnInfo, FALSE, "Good guys has win 3 rouonds.");

```

Comment:
最好还是写成用 __leave 来离开作用域
```suggestion
        if (pRoom->Game_Win>=3)
        {
            bSuccess = ReplyPlayerSelectTeam(pConnInfo, FALSE, "Good guys has win 3 rouonds.");
            __leave;
        }
```

Reply:
写成用 __leave 来离开作用域